### PR TITLE
Add UBSan support for C family languages

### PR DESCRIPTION
**Pull request by rahul-malik**
Please test with following pull request:
https://github.com/apple/swift/pull/18553

### CODE REVIEW

File: Sources/Build/Sanitizers.swift [hunk 1]
```diff
@@ -39,17 +45,18 @@ public struct EnabledSanitizers {
...
     public func compileCFlags() -> [String] {
-        return sanitizers.map({ "-fsanitize=\($0.rawValue)" })
+        return sanitizers.map{ $0.flagName }
     }
 
     /// Sanitization flags for the Swift compiler.
     public func compileSwiftFlags() -> [String] {
-        return sanitizers.map({ "-sanitize=\($0.rawValue)" })
+        // UBSan is only available for C family languages
+        return sanitizers.filter{ $0 != .undefined }.map{ $0.flagName }
```
Comment (keith) [inline Sources/Build/Sanitizers.swift]: this used to return `-sanitize=asan` and now is `-fsanitize=asan`, that's intentional?

Comment (rahul-malik) [reply Sources/Build/Sanitizers.swift]: Wait, this is an old commit, one sec

**Comment by rahul-malik** [comment]
@keith - Forgot to push my more recent changes, updated now!

File: Sources/Build/Sanitizers.swift (hunk 1 above)
Comment (rahul-malik) [reply Sources/Build/Sanitizers.swift]: @keith - thats actually not correct though, before it was `-sanitize=address` and the `asan`/`tsan` values were used to create the dylib path to the santizer runtime

File: Fixtures/CFamilyTargets/CUBSan/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# CUBSan
+
+A description of this package.
```
Comment (vlm) [inline Fixtures/CFamilyTargets/CUBSan/README.md]: Hm? ;)

File: Fixtures/CFamilyTargets/CUBSan/include/main.h [hunk 3]
```diff
@@ -0,0 +1 @@
+int main(int argc, char **argv);
```
Comment (aciidb0mb3r) [inline Fixtures/CFamilyTargets/CUBSan/include/main.h]: The header is not required.

Comment (rahul-malik) [reply Fixtures/CFamilyTargets/CUBSan/include/main.h]: Yeah I agree, if anything this seems like the linking testing is happening in the Swift PR and should be sufficient. Should we just remove the fixture entirely then?

**Comment by rahul-malik** [comment]
@acdlite @keith @vlm - Thanks for your feedback, I've removed the fixture entirely since it isn't necessary for this PR. There are tests for the Swift driver changes that assert the runtime is linked correctly which should be sufficient.

**Comment by aciidb0mb3r** [comment]
Cool, thanks!

**Comment by aciidb0mb3r** [comment]
I'll hold off merging until https://github.com/apple/swift/pull/18553 is merged

**Comment by aciidb0mb3r** [comment]
@swift-ci smoke test

**Comment by rahul-malik** [comment]
@aciidb0mb3r - Finally got back to this. I believe I figured out the test issues and updated my swift pr here https://github.com/apple/swift/pull/18553. Feel free to re-test

**Comment by aciidb0mb3r** [comment]
Thanks!

**Pull request merged**